src/panels/config/script/ha-script-editor.ts

```diff
@@ -160,14 +160,16 @@ export class HaScriptEditor extends KeyboardShortcutMixin(LitElement) {
       return nothing;
     }
 
+    const useBlueprint = "use_blueprint" in this._config;
+
     const schema = this._schema(
       !!this.scriptId,
-      "use_blueprint" in this._config,
+      useBlueprint,
       this._config.mode
     );
 
     const data = {
-      mode: MODES[0],
+      ...(!useBlueprint && { mode: MODES[0] }),
```

Comment:
```suggestion
      ...(!this._config.mode && !useBlueprint && { mode: MODES[0] }),
```

Reply:
May I ask why this change? I'm not seeing it. 

this._config spreads over the mode later in the assignment, so this seems like this suggestion will have no effect. I have no objection to taking it, but was just curious.

Reply:
Mainly because it confused me at first, and to micro optimise 😄

I think it might be ever clearer to just do:

```ts
const data = useBlueprint ? {blueprint_stuff, ...this._config} : {noneblueprint_stuff, ...this._config};
data.id = this._entityId;
```

Reply:
I just took your original suggestion, hope that's ok. I consider this other one but that ends up duplicating a lot of the lines, since the only difference between blueprint_stuff and non_blueprint_stuff is the mode key. Otherwise have to duplicate things like the max assignment.

Reply:
I dont think a blueprint automation should have a max option?

Reply:
Hmm, well it currently does. IDK if it should or not.